env/file_system.cc

```diff
@@ -115,6 +116,25 @@ IOStatus FileSystem::ReuseWritableFile(const std::string& fname,
   return NewWritableFile(fname, opts, result, dbg);
 }
 
+IOStatus FileSystem::NewLogger(const std::string& fname,
+                               const IOOptions& io_opts,
+                               std::shared_ptr<Logger>* result,
+                               IODebugContext* dbg) {
```

Comment:
If this is the desired behavior for all FileSystems, then the default for Windows should also be changed.

If it is not the default for Windows, the NewLogger method should continue to be abstract and this code moved into fs_posix